src/board.rs

```diff
@@ -106,123 +215,175 @@ impl Chessboard {
-    //             if beta <= alpha
-    //                 break
-    //         return max_eval
-    //     else
-    //         min_eval = +infinity
-    //         for each child of position
-    //         eval = minimax(child, depth - 1, alpha, beta, true)
-    //         min_eval = min(min_eval, eval)
-    //         beta = min(beta, eval)
-    //         if beta <= alpha
-    //             break
-    //     return min_eval
-    // }
-}
-//Below here is just ideas for functions:
+                for &(piece, mask) in self.get_pieces().iter() {
+                    if (mask >> square_ndx) & 1 != 0 {
+                        if empty_squares > 0 { 
+                            row_string.push_str(&empty_squares.to_string());
+                        }
+                        row_string.push(piece);
+                        empty_squares = 0;
+                    }
+                }
+            }
 
-//fn what_occupy square() {
-//     check each piece type with the square and see which returns true
-// }
+            // Append the count of empty squares at the end of the row string
+            if empty_squares > 0 {
+                row_string.push_str(&empty_squares.to_string());
+            }
 
-// fn is_square_occupied(square) {
-// bitwise and every piece together
-// search through the new u64 and see where there are 1s
-// return the indices of the 1s
-// }
+            // Add the row string to the FEN components vector
+            string_array.push(row_string);
+            string_array.push("/".to_owned());
+        }
 
-// en_passant_square() {
-//     // Return the square where an en passant capture is possible, or -1 if not possible
-//     // For example, if the last move was a double-step pawn move, return the square where the opponent's pawn can capture en passant.
-// }
+        let binding = string_array.concat();
+        fen_string = binding.chars();
+        fen_string.next_back();
+        let fen_string_no_slash = fen_string.as_str();
+        for item in &mut string_array {
+            item.clear();
+        }
+        string_array.push(fen_string_no_slash.to_owned());
 
-// castling_rights() {
-//     // Use flags or bitmasks to represent castling rights for both players
-//     // For example, if kingside castling is allowed for white, set white_kingside_castle = true
-//     // Similarly, maintain flags for black's castling rights
-// }
+        string_array.push(" ".to_owned());
 
-// make_move(move) {
-//     // Update the bitboards and game state based on the move
-//     // For example, if it's a pawn move, update the pawn bitboard
-//     // If it's a capture, clear the bit for the captured piece
-//     // Update en passant square and castling rights accordingly
-//     // Switch turns: white_turn = !white_turn;
-// }
+        // Whose turn
+        string_array.push(if self.white_turn {
+            "w ".to_string()
+        } else {
+            "b ".to_string()
+        });
```

Comment:
this can be minorly simplified
`if self.white_turn { "w " } else { "b "}.to_string()`